From d6c333203189438d1a7ca1c568b8cb0d7d850f45 Mon Sep 17 00:00:00 2001
From: Yuriy Umanets <yuriy_umanets@xyratex.com>
Date: Sun, 10 Apr 2011 07:12:46 -0600
Subject: [PATCH 055/177] - no root_fid is needed in c2_md_store_init(). It
 always can be found as mkfs should create it with nskey (C2_MD_ROOT_FID,
 "ROOT").

---
 mdservice/md_foms.c  |  3 +++
 mdstore/mdstore.c    | 15 ++++++++-------
 mdstore/mdstore.h    |  5 ++---
 utils/mkfs.colibri.c |  2 +-
 4 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/mdservice/md_foms.c b/mdservice/md_foms.c
index 4fecd92..25971ca 100644
--- a/mdservice/md_foms.c
+++ b/mdservice/md_foms.c
@@ -539,6 +539,9 @@ static int c2_md_readdir_fom_perm(struct c2_fom *fom)
         return 0;
 }
 
+/**
+   Number of entries preallocated for readdir.
+ */
 #define C2_MD_READDIR_ENTRIES_ALLOC 4096
 
 static int c2_md_readdir_fom_state(struct c2_fom *fom)
diff --git a/mdstore/mdstore.c b/mdstore/mdstore.c
index 24ec404..094d1f8 100644
--- a/mdstore/mdstore.c
+++ b/mdstore/mdstore.c
@@ -32,21 +32,20 @@ static const struct c2_addb_loc mdstore_addb_loc = {
 };
 
 struct c2_fid C2_MD_ROOT_FID = {
-        .f_container = 1024ULL, 
+        .f_container = 1ULL, 
         .f_key       = 1ULL
 };
 
 int c2_md_store_init(struct c2_md_store *md, 
                      struct c2_cob_domain_id *id,
                      struct c2_dbenv *db, 
-                     struct c2_fid *root_fid, 
                      int init_root)
 {
+        struct c2_cob_nskey   *nskey;
         struct c2_db_tx        tx;
         int                    rc;
 
-        C2_ASSERT(md != NULL && id != NULL && 
-                  db != NULL && root_fid != NULL);
+        C2_ASSERT(md != NULL && id != NULL && db != NULL);
         
         C2_SET0(md);
         c2_addb_ctx_init(&md->md_addb, &mdstore_addb_ctx, 
@@ -60,8 +59,10 @@ int c2_md_store_init(struct c2_md_store *md,
         }
         if (init_root) {
                 c2_db_tx_init(&tx, db, 0);
-                rc = c2_md_store_locate(md, root_fid, &md->md_root, 
-                                        C2_MD_STORE_LOCATE_STORED, &tx);
+                
+                c2_cob_make_nskey(&nskey, &C2_MD_ROOT_FID, "ROOT", 4);
+                rc = c2_cob_lookup(&md->md_dom, nskey, CA_NSKEY_FREE, 
+                                   &md->md_root, &tx);
                 if (rc) {
                         c2_db_tx_abort(&tx);
                 } else {
@@ -744,6 +745,6 @@ int c2_md_store_alloc(struct c2_md_store *md,
         
         rc = c2_cob_alloc(&md->md_dom, cob);
         if (rc == 0)
-                (*cob)->co_nsrec.cnr_fid = *fid;
+                (*cob)->co_fid = *fid;
         return rc;
 }
diff --git a/mdstore/mdstore.h b/mdstore/mdstore.h
index ee20ac3..c004568 100644
--- a/mdstore/mdstore.h
+++ b/mdstore/mdstore.h
@@ -59,13 +59,12 @@ enum c2_md_valid_flags {
 extern struct c2_fid C2_MD_ROOT_FID;
 
 /**
-   Init mdstore and get it ready to work. Specified root_fid
-   is used to locate root cob init_root == !!1.
+   Init mdstore and get it ready to work. Ifinit_root == !!1
+   then root cob is initialized.
 */
 int c2_md_store_init(struct c2_md_store *md, 
                      struct c2_cob_domain_id *id,
                      struct c2_dbenv *db, 
-                     struct c2_fid *root_fid, 
                      int init_root);
 
 /**
diff --git a/utils/mkfs.colibri.c b/utils/mkfs.colibri.c
index be31c16..cbb209d 100644
--- a/utils/mkfs.colibri.c
+++ b/utils/mkfs.colibri.c
@@ -112,7 +112,7 @@ int main(int argc, char *argv[])
 	if (rc)
 	        goto out_free_c2;
 	        
-        rc = c2_md_store_init(&md, &cdid, &db, &C2_MD_ROOT_FID, 0);
+        rc = c2_md_store_init(&md, &cdid, &db, 0);
         if (rc)
                 goto out_free_db;
 
-- 
1.8.3.2

